### ISSUE TITLE
Bayesian Optimization for auto-tuning parameters

### ISSUE THREAD

**Issue by thomasfrederikhoeck** [andersbogsnes/ml_tooling#165]
#### Code Sample

```python
search_space = {C : [0,100]}
model = IrisModel(LogisticRegression())
model.bayesian_optim(search_space)

```
#### Problem description
It would be very helpful to use bayesian optimization instead of 'idiotic' grid-searching. The package BayesianOptimization has the need tools and the following can be used as inspiration:

https://github.com/fmfn/BayesianOptimization/blob/master/examples/sklearn_example.py

Checklist
---------
- [ ] I have updated ml-tooling to the newest version `pip install ml-tooling --upgrade`
- [ ] I have checked that this issue does not already exist

**Comment by thomasfrederikhoeck** [comment]
Other optimizers are available but they all seem to depend on MongoDB which might be overkill.

**Comment by andersbogsnes** [comment]
Yes, it would be a nice addition. There's also an sklearn wrapper for [hyperopt](https://github.com/hyperopt/hyperopt-sklearn) which uses mongodb for parallel searches (maybe the one you were thinking of?), but runs serially without.

**Comment by thomasfrederikhoeck** [comment]
As I see it  hyperopt-sklearn is meant to be used instead the sklearn estimator that you want to be tune. However hyperopt is a good suggestion for tuning. Note that

> Currently two algorithms are implemented in hyperopt:
> 
>  - Random Search
>  - Tree of Parzen Estimators (TPE)
> 
> Hyperopt has been designed to accommodate Bayesian optimization algorithms based on Gaussian > processes and regression trees, but these are not currently implemented.

 -http://hyperopt.github.io/hyperopt/

**Comment by andersbogsnes** [comment]
closed by #635
